src/dune_rules/install_rules.ml

```diff
@@ -153,47 +153,50 @@ end = struct
       in
       let virtual_library = Library.is_virtual lib in
       let modules =
-        let common m =
```

Comment:
should we just forbid public_name on melange libraries instead? seems simpler

Reply:
that would make impossible to use melange in projects with shared libraries, where some dependent on the shared library is public (e.g. Ahrefs).

Reply:
Okay, then let's work towards proper support for public libraries then. The situation you're describing isn't going to work correctly when the public library is installed anyway.

Reply:
The libraries are not installed in Ahrefs case, afaik, we only use `public_name` to get them listed on the main index page of odoc, for internal documentation purposes.

Reply:
How about we try adding an index that includes private libraries then? Seems like a more useful solution and probably simpler too.